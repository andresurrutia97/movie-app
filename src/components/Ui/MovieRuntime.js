import React from 'react';
import PropTypes from 'prop-types';

const MovieRuntime = ({ runtime }) => (
  <div className="w-auto inline-block rounded-full custom-pink-bg py-1 px-3 my-3">
    {`${runtime} min`}
  </div>
);

MovieRuntime.propTypes = {
  runtime: PropTypes.number,
};

MovieRuntime.defaultProps = {
  runtime: 0,
};

export default MovieRuntime;
