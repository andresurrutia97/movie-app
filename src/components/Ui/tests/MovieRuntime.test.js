import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MovieRuntime from '../MovieRuntime';

describe('<MovieRuntime/>', () => {
  const component = <MovieRuntime runtime={123} />;

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text equal to "123 min"', () => {
    const wrapper = shallow(component);
    expect(wrapper.props().children).toBe('123 min');
  });
});
