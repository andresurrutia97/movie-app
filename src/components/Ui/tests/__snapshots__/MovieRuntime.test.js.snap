// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieRuntime/> should render correctly  1`] = `
<div
  className="w-auto inline-block rounded-full custom-pink-bg py-1 px-3 my-3"
>
  123 min
</div>
`;
